drivers/watchdog/wdt_nrfx_wdt.c

```diff
@@ -0,0 +1,213 @@
+ */
+static wdt_callback_t      m_callbacks[NRF_WDT_CHANNELS_NUMBER];
+
+/*
+ * The m_allocated_channels variable stores number of currently allocated
+ * and activated watchdog channels.
+ */
+static u8_t                m_allocated_channels;
+
+/*
+ * The m_timeout variable stores watchdog timeout value in millisecond units.
+ * It is used to check whether installing watchdog provide the same timeout
+ * value in the window configuration.
+ */
+static u32_t               m_timeout;
+
+
+static void wdt_event_handler(void)
+{
+	int i;
+
+	for (i = 0; i < m_allocated_channels ; ++i) {
+		if (nrf_wdt_request_status(m_ch[i])) {
+			if (m_callbacks[i]) {
+				m_callbacks[i](DEVICE_GET(wdt_nrf), i);
+			}
+		}
+	}
+}
+
+static void wdt_nrf_isr(struct device *dev)
+{
+	ARG_UNUSED(dev);
+	/* We need to implement our own interrupt handler until nrfx one will
+	 * be fixed.
+	 * Clearing events also clears the reload register and has to be done
+	 * AFTER event handler.
+	 */
+	wdt_event_handler();
+	nrf_wdt_event_clear(NRF_WDT_EVENT_TIMEOUT);
+}
+
+static int wdt_nrf_setup(struct device *dev, u8_t options)
+{
+	nrfx_wdt_config_t config;
+	nrfx_err_t err_code;
+
+	ARG_UNUSED(dev);
+
+	if (!nrf_clock_lf_is_running()) {
+		return -ENOTSUP;
+	}
+
+	/* Activate all available options. Run in all cases. */
+	config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_HALT;
+
+	/* Deactivate running in sleep mode. */
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP;
+	}
+
+	/* Deactivate running when debugger is attached. */
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT;
+	}
+
+	config.reload_value = m_timeout;
+
+	err_code = nrfx_wdt_init(&config, wdt_event_handler);
+	if (err_code == NRFX_ERROR_INVALID_STATE) {
+		return -EBUSY;
+	}
+
+	IRQ_CONNECT(CONFIG_WDT_NRF_IRQ,
+		    CONFIG_WDT_NRF_IRQ_PRI, wdt_nrf_isr,
+		    DEVICE_GET(wdt_nrf), 0);
+	irq_enable(CONFIG_WDT_NRF_IRQ);
+
+	nrfx_wdt_enable();
+
+	return 0;
+}
+
+static int wdt_nrf_disable(struct device *dev)
+{
+	/* Started watchdog cannot be stopped on nRF devices. */
+	ARG_UNUSED(dev);
+	return -EPERM;
+}
+
+static int wdt_nrf_install_timeout(struct device *dev,
+				   const struct wdt_timeout_cfg *cfg)
+{
+	nrfx_err_t err_code;
+
+	ARG_UNUSED(dev);
+
+	if (cfg->flags != WDT_FLAG_RESET_SOC) {
+		return -ENOTSUP;
+	}
+
+	if (cfg->window.min != 0) {
+		return -EINVAL;
+	}
+
+	if (m_allocated_channels == 0) {
+		/* Save timeout value from first registered watchdog channel. */
+		m_timeout = cfg->window.max;
+	} else if (cfg->window.max != m_timeout) {
+		return -EINVAL;
+	}
+
+	err_code = nrfx_wdt_channel_alloc(&m_ch[m_allocated_channels]);
+
+	if (err_code == NRFX_ERROR_NO_MEM) {
+		return -ENOMEM;
+	}
+
+	if (cfg->callback != NULL) {
+		m_callbacks[m_allocated_channels] = cfg->callback;
+	}
+
+	m_allocated_channels++;
+	return (m_allocated_channels - 1);
```

Comment:
Perhaps it would be more readable to use a local variable `channel_id`, initialize it to `m_allocated_channels` and return its value here.